bench/Cargo.toml

```diff
@@ -5,7 +5,7 @@ version = "0.1.0"
 authors = ["The Rust Project Developers"]
 license = "MIT/Apache-2.0"
 repository = "https://github.com/rust-lang/regex"
-documentation = "http://doc.rust-lang.org/regex/regex_syntax/index.html"
+documentation = "http://doc.rust-lang.org/regex/regex/index.html"
```

Comment:
This is just a random nit I had when reading the cargo file. It seems odd for the benchmark suite to point to the syntax crate for documentation. I'm not at all attached to this change.